packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -145,34 +147,132 @@ describe('Context bindings - Injecting dependencies of classes', () => {
   });
 
   it('injects a getter function', async () => {
-    ctx.bind('key').to('value');
+    ctx.bind('hash').to('123');
 
     class Store {
-      constructor(@inject.getter('key') public getter: Getter<string>) {}
+      constructor(@inject.getter('hash') public getter: Getter<string>) {}
     }
 
     ctx.bind('store').toClass(Store);
     const store = ctx.getSync<Store>('store');
 
     expect(store.getter).to.be.Function();
-    expect(await store.getter()).to.equal('value');
+    expect(await store.getter()).to.equal('123');
 
     // rebind the value to verify that getter always returns a fresh value
-    ctx.bind('key').to('new-value');
-    expect(await store.getter()).to.equal('new-value');
+    ctx.bind('hash').to('456');
+    expect(await store.getter()).to.equal('456');
+  });
+
+  describe('in SINGLETON scope', () => {
+    it('throws if a getter cannot be resolved by the owning context', async () => {
+      class Store {
+        constructor(@inject.getter('hash') public getter: Getter<string>) {}
+      }
+
+      // Create a child context of `ctx`
+      const requestCtx = new Context(ctx, 'child');
+      // Bind `hash` to `value`
+      requestCtx.bind('hash').to('123');
+
+      // Bind `Store` as a singleton
+      ctx
+        .bind('store')
+        .toClass(Store)
+        .inScope(BindingScope.SINGLETON);
+
+      // Create the store instance from the child context
+      // Now the singleton references a `getter` to `binding`
+      const store = requestCtx.getSync<Store>('store');
+
+      await expect(store.getter()).to.be.rejectedWith(
+        /The key 'hash' is not bound to any value in context .+/,
+      );
+
+      // bind the value to to ctx
+      ctx.bind('hash').to('456');
+      expect(await store.getter()).to.equal('456');
+    });
+
+    it('throws if a value cannot be resolved by the owning context', async () => {
+      class Store {
+        constructor(@inject('hash') public hash: string) {}
+      }
+
+      // Create a child context of `ctx`
+      const requestCtx = new Context(ctx, 'child');
+      // Bind `hash` to `value`
+      requestCtx.bind('hash').to('123');
```

Comment:
Sorry, what does binding key `has` to `123` in the child context signify to the test?

Reply:
It verifies that the binding in a child context does not fulfill the dependency for a singleton at parent level.